### PR TITLE
Ora -e

### PR DESCRIPTION
**Pull request by efischer19**
Taking a stab at this, need a branch up on github for testing.

### CODE REVIEW

**Comment by edx-webhook** [comment]
## PR Cover Letter
- [ ] A detailed description of the changes in the body of the PR
- [ ] If available, a link to the relevant JIRA ticket in the body of the PR
  - `[PROJ-XXXX](https://openedx.atlassian.net/browse/PROJ-XXXX)`

## Sandbox
- [ ] Build a sandbox for your branch and add a link here

## Testing
- [ ] i18n
- [ ] RTL
- [ ] Unit, integration, acceptance tests as appropriate
- [ ] Analytics
- [ ] Performance
- [ ] Database migrations are backwards-compatible

## Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] Code review #1
- [ ] Code review #2
- [ ] Docs review (required for UI strings/error messages)
- [ ] UX review
- [ ] Accessibility review
- [ ] Product review

## Documentation considerations
- [ ] Are there user visible changes? If so, list who is affected, and include a pointer to the relevant code.
- [ ] Should this be mentioned in the release notes? If so, make sure that you included a description.

## DevOps assistance
If it's needed, or even if you suspect it might be needed, do any/all these things as early as possible. Don't surprise devops right before merging! For example, touching envs/aws.py or or envs/common.py are dead giveaways that you'll need assistance when pushing your changes out. Things that require devops attention include, but are not limited to:
- configuration changes
- feature flags
- migrations
Ways to get devops attention, pick-and-choose whichever is most appropriate for your specific PR:
- [ ] Tag devops on your PR (`@edx/devops`), either for review (more specific to devops = better) or just a general FYI
- [ ] File a support ticket

FYI: Tag anyone who might be interested in this PR here.

## Post-review
- [ ] Squash commits

**Comment by efischer19** [comment]
This seems to be working, all manual testing of ORA on my local devstack and sandbox https://efischer19.sandbox.edx.org/ seems to be working, as are acceptance tests. I had to add some waits to get acceptance tests passing - I'm not sure if that is caused by something related to this install or not.

Some notes on sandbox installation - in order to get ora2 installed properly, I had to `sudo su edxapp -s /bin/bash`, navigate to the `/edx/app/edxapp/` directory, `source venvs/edxapp/bin/activate`, and use pip. Initially, a different version than I expected was installed, with a `-e` flag. There was code related to this version stored in `venvs/edxapp/src`. I ended up using pip to uninstall that version, deleted the directory full of code, and re-installed using `pip install git+https://github.com/edx/edx-ora2.git@christina/ora-e#egg=ora2`. My devstack was built from scratch with this branch (https://github.com/edx/edx-platform/pull/11623), and the output of `pip freeze | grep ora2` was what I expected.

One remaining question - do we need all of the files that are currently being included? A full list is here: https://gist.github.com/efischer19/a9b7188ac7f8fb6cbcf3
I think we could remove a lot of test files, and maybe the contents of `openassessment/xblock/static/sass/xb/libs/`, but I'm not 100% on whether `*.scss` files should be included. Whatever ends up being left out just needs to be specified via an `exclude`, `recursive-exclude`, or `prune` command in `MANIFEST.in`

@nedbat and @dianakhuang, can you review this when you get a chance? Also FYI @cahrens.

**Comment by efischer19** [comment]
After some trial-and-erroring, I believe this is now using a minimal install. New file list visible at https://gist.github.com/efischer19/9b87e103e8983f35381e. I'll get my sandbox updated shortly.

**Comment by nedbat** [comment]
@efischer19 thanks for doing this. fwiw, there's no real reason to aim for a minimal install, so long as it has all the files it needs.

**Comment by efischer19** [comment]
Right on @nedbat, thanks.

@cahrens - I've updated my sandbox at https://efischer19.sandbox.edx.org/ and all acceptance tests are passing. I'll wait for your thumbs to merge.

File: MANIFEST.in [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-recursive-include apps/openassessment/templates *.html
-recursive-include apps/openassessment/xblock/static *
-recursive-include apps/openassessment/locale *.po
-recursive-include apps/openassessment/locale *.mo
 include LICENSE
 include AUTHORS
 include README.rst
+include openassessment/xblock/static/css/*.css
+include openassessment/xblock/static/js/openassessment*.min.js
+recursive-include openassessment/xblock/static/js/src *.js
```
Comment (cahrens) [inline MANIFEST.in]: Do we need the non-minified JS?

File: setup.py [hunk 2]
```diff
@@ -60,7 +45,8 @@ def load_requirements(*requirements_paths):
         'Operating System :: OS Independent',
         'Programming Language :: Python',
     ],
-    packages=PACKAGES,
+    packages=find_packages(exclude=['*.test', '*.tests']),
```
Comment (cahrens) [inline setup.py]: Is specifying the exclusion of tests here actually don't anything? I'm guessing not (I think the paths have to be exact). If the exclusion is happening through MANIFEST.in, let's remove it here.

File: MANIFEST.in (hunk 1 above)
Comment (efischer19) [inline MANIFEST.in]: I tried to load without, and was getting errors
https://github.com/edx/edx-ora2/pull/882#issuecomment-188871357

**Comment by cahrens** [comment]
Just a couple of questions/simplifications, otherwise :+1:

File: MANIFEST.in (hunk 1 above)
Comment (cahrens) [inline MANIFEST.in]: So it seems there are JS files that we access besides the minified files. I wasn't aware of that.

Comment (cahrens) [inline MANIFEST.in]: Seems like it is in DEBUG mode. Given that we don't need to have the absolutely smallest number of files, it seems fine.

if settings.DEBUG:
            self.add_javascript_files(fragment, "static/js/src/oa_shared.js")
            self.add_javascript_files(fragment, "static/js/src/oa_server.js")
            self.add_javascript_files(fragment, "static/js/src/studio")
        else:
            # TODO: switch to add_javascript_url once XBlock resources are loaded from the CDN
            fragment.add_javascript(pkg_resources.resource_string(__name__, "static/js/openassessment-studio.min.js"))

File: setup.py (hunk 2 above)
Comment (efischer19) [inline setup.py]: Just verified, and without any `exclude`, we get test packages included, and `blahblah/test/__init__.py` files are present.

File: MANIFEST.in (hunk 1 above)
Comment (efischer19) [inline MANIFEST.in]: Ah cool, that makes sense.

I could see someone installing and wanting to run DEBUG mode to trace down a js error, so this seems reasonable.

File: setup.py (hunk 2 above)
Comment (cahrens) [inline setup.py]: Cool, thanks for following up. I'm content with this PR now.

**Pull request merged**